src/pfe/file-watcher/server/test/functional-test/suites/project/tests/project-specification.ts

```diff
@@ -0,0 +1,372 @@
+                            clearInterval(timer);
+                            return resolve();
+                        }
+                    }
+                }, timeoutConfigs.defaultInterval);
+            });
+
+            if (eventFound && event) {
+                expect(event);
+                expect(event.eventName);
+                expect(event.eventName).to.equal(targetEvent);
+                expect(event.eventData);
+                expect(event.eventData["operationId"]);
+                expect(event.eventData["projectID"]);
+                expect(event.eventData["projectID"]).to.equal(projData.projectID);
+                expect(event.eventData["status"]);
+                expect(event.eventData["status"]).to.equal("failed");
+                expect(event.eventData["error"]);
+                expect(event.eventData["error"]).to.equal(`BAD_REQUEST: Each setting must have a name and a value.`);
+            } else {
+                fail(`failed to find ${targetEvent} for project specific setting`);
+            }
+        }).timeout(timeoutConfigs.defaultTimeout);
+
+        _.forEach(combinations, async (combo) => {
+            const setting = combo["setting"];
+            let value = combo["value"];
+
+            if (setting === "internalPort") {
+                const projectInfo = _.cloneDeep(projData) as ProjectInfo;
+                projectInfo["autoBuildEnabled"] = true;
+                projectInfo["startMode"] = "run";
+                if (process.env.IN_K8) {
+                    const internalPorts = await getApplicationContainerInfoInK8(projectInfo, new Operation("general", projectInfo));
+                    value = internalPorts.podPorts[Math.floor(Math.random() * internalPorts.podPorts.length)];
+                } else {
+                    const internalPorts = await getApplicationContainerInfo(projectInfo, await projectUtil.getContainerName(projectInfo));
+                    value = internalPorts.containerPorts[Math.floor(Math.random() * internalPorts.containerPorts.length)];
+                }
+            }
+            if (setting === "internalDebugPort") {
+                if (process.env.IN_K8) return; // internal debug port setting is not supported in kube
+                const exposedDebugPorts = project_configs.exposedDebugPorts[projData.projectType];
+                value = exposedDebugPorts[Math.floor(Math.random() * exposedDebugPorts.length)];
+            }
+            if (setting === "mavenProfiles" || setting === "mavenProperties") {
+                if (!project_configs.mavenProfileCapabilities[projData.projectType]) {
+                    value = [];
+                    combo["eventKeys"].push("error");
+                }
+                // for spring and liberty project types support maven profiles we need to set the value here
+            }
+
+            it(`config project specification settings ${setting} to ${value}`, async () => {
```

Comment:
![Screen Shot 2019-08-12 at 5 17 39 PM](https://user-images.githubusercontent.com/15131537/62899148-301a9980-bd25-11e9-9eb4-65fc30275b1f.png)

Can you see the project setting test case for `internalPort` has been triggered when you run the tests? I can't find it when I run the tests

Reply:
Should be fixed now.